src/integrationTest/java/org/opensearch/security/http/OnBehalfOfJwtAuthenticationTest.java

```diff
@@ -0,0 +1,136 @@
+	private static final String encryptionKey = Base64.getEncoder().encodeToString("encryptionKey".getBytes(StandardCharsets.UTF_8));
+
+	private static final OnBehalfOfJwtAuthorizationHeaderFactory tokenFactory = new OnBehalfOfJwtAuthorizationHeaderFactory(
+			signingKey,
+			issuer,
+			subject,
+			audience,
+			roles,
+			backendRoles,
+			expirySeconds,
+			headerName,
+			encryptionKey
+	);
+
+	public static final String ADMIN_USER_NAME = "admin";
+	public static final String DEFAULT_PASSWORD = "secret";
+	public static final String OBO_TOKEN_REASON = "{\"reason\":\"Test generation\"}";
+	public static final String OBO_ENDPOINT_PREFIX = "_plugins/_security/api/user/onbehalfof";
+
+	@ClassRule
+	public static final LocalCluster cluster = new LocalCluster.Builder()
+			.clusterManager(ClusterManager.SINGLENODE).anonymousAuth(false)
+			.users(ADMIN_USER)
+			.nodeSettings(Map.of(
+					"plugins.security.allow_default_init_securityindex", true,
+					"plugins.security.restapi.roles_enabled", List.of("user_admin__all_access")
+			))
+			.authc(AUTHC_HTTPBASIC_INTERNAL)
+			.onBehalfOf(new OnBehalfOfConfig().signing_key(signingKey).encryption_key(encryptionKey))
+			.build();
+
+	@Test
+	public void shouldAuthenticateWithOBOToken() {
+		try(TestRestClient client = cluster.getRestClient(tokenFactory.generateValidToken())){
+
+			TestRestClient.HttpResponse response = client.getAuthInfo();
+
+			response.assertStatusCode(200);
+			String username = response.getTextFromJsonBody(POINTER_USERNAME);
+			assertThat("testUser", equalTo(username));
+		} catch (Exception e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	@Test
+	public void shouldAuthenticateWithOBOTokenEndPoint() {
+		Header adminOboAuthHeader;
+
+		try (TestRestClient client = cluster.getRestClient(ADMIN_USER_NAME, DEFAULT_PASSWORD)) {
+
+			client.assertCorrectCredentials(ADMIN_USER_NAME);
+
+			TestRestClient.HttpResponse response = client.postJson(OBO_ENDPOINT_PREFIX, OBO_TOKEN_REASON);
+			response.assertStatusCode(200);
+
+			Map<String, Object> oboEndPointResponse = response.getBodyAs(Map.class);
+			assertThat(oboEndPointResponse, allOf(
+					aMapWithSize(3),
+					hasKey("user"),
+					hasKey("onBehalfOfToken"),
+					hasKey("duration")));
+
+			String encodedOboTokenStr = oboEndPointResponse.get("onBehalfOfToken").toString();
+			System.out.println("This is the OBO Token of admin user: " + encodedOboTokenStr);
+
+			adminOboAuthHeader = new BasicHeader("Authorization", "Bearer " + encodedOboTokenStr);
+		}
+
+		try (TestRestClient client = cluster.getRestClient(adminOboAuthHeader)) {
+
+			TestRestClient.HttpResponse response = client.getAuthInfo();
+			response.assertStatusCode(200);
+
+			String username = response.getTextFromJsonBody(POINTER_USERNAME);
+			assertThat(username, equalTo(ADMIN_USER_NAME));
+		}
+	}
```

Comment:
We should also have tests that check that OBO tokens cannot be used to 
- Get a new OBO token
- Reset the password (?)

Note; these could be follow up tasks

Reply:
Thanks for the reminder! I have created an issue to track these edge cases (https://github.com/opensearch-project/security/issues/2891)